worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -227,6 +230,7 @@ class FetchHandler(
           }
           val meta = fileInfo.getFileMeta.asInstanceOf[ReduceFileMeta]
           if (readLocalShuffle) {
+            streams.put(-1, (shuffleKey, fileName))
```

Comment:
QQ: Is `streamId` expected to be `-1` when it is local (if yes, use that instead) ? Trying to understand how it is getting cleaned up.

Reply:
@mridulm,  you could refer to `replyStreamHandler(client, rpcRequestId, -1, meta.getNumChunks, isLegacy, meta.getChunkOffsets, fileInfo.getFilePath)` for reading local shuffle, which replies with `streamId` -1. I only keep the same `streamId` when reading local shuffle.

Reply:
As readLocalShuffle may have concurrent calls, if we use -1 as the key, then the streams will ultimately only retain one result, NPE still remains when handleEndStream.

Reply:
I have similar query as @RexXiong - thanks for elaborating it better Shuang !